titus-server-federation/src/main/java/com/netflix/titus/federation/service/AggregatingJobServiceGateway.java

```diff
@@ -135,15 +138,24 @@ public AggregatingJobServiceGateway(GrpcConfiguration grpcConfiguration,
         }
         JobManagementServiceStub client = optionalClient.get();
 
-        JobDescriptor withStackName = addStackName(jobDescriptor);
+        JobDescriptor enrichedJobDescriptor;
+        if (federationConfiguration.isFederationJobIdCreationEnabled()) {
+            String federatedJobId = UUID.randomUUID().toString();
+            enrichedJobDescriptor = addJobAttributes(jobDescriptor,
+                    ImmutableMap.of(JOB_ATTRIBUTES_STACK, federationConfiguration.getStack(),
```

Comment:
logic being duplicated from `addStackName()`

Reply:
We are avoiding calling `build()`. What do you propose?

Reply:
re-shuffling things a bit to avoid calling `build()` in `addStackName()` too early might be enough. It seems we always want to add the stackName, so my suggestion would be to factor it out of the if/else chain and always add it unconditionally keeping the `Builder` alive by calling `.build()` as late as possible.